### PR TITLE
ja_serializer also serializes nested attributes

### PR DESCRIPTION
**Pull request by broodfusion**
Hi @beerlington , finally got around to submitting a PR for the issue I opened 2 months ago https://github.com/vt-elixir/ja_serializer/issues/290
Please take a look at it and let me know your thoughts.
Thanks!

### CODE REVIEW

**Comment by broodfusion** [comment]
BTW, I didn't touch any part of the code that's failing the test

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ erl_crash.dump
 /doc
 /docs
 /bench/snapshots
+.elixir_ls/
```
Comment (moxley) [inline .gitignore]: Or, you can ignore `.elixir_ls` outside of this repo by configuring a global ignore file: http://egorsmirnov.me/2015/05/04/global-gitignore-file.html

File: lib/ja_serializer/formatter/utils.ex [hunk 2]
```diff
@@ -25,25 +26,32 @@ defmodule JaSerializer.Formatter.Utils do
...
     Map.put(accumulator, format_key(key), deep_format_keys(value))
   end
+
   defp deep_format_key_value({key, value}, accumulator) do
     Map.put(accumulator, format_key(key), value)
   end
 
   @doc false
-  def format_key(k) when is_atom(k), do: k |> Atom.to_string |> format_key
+  def format_key(k) when is_atom(k), do: k |> Atom.to_string() |> format_key
```
Comment (moxley) [inline lib/ja_serializer/formatter/utils.ex]: Since you're adding parentheses here, can you add them to the `format_key()` call too?

**Comment by moxley** [comment]
@broodfusion: I get the same test failures off your branch that are reported in Travis, but not when running of master:

```
  1) test map formatter can be changed (JaSerializer.Formatter.AttributeTest)
     test/ja_serializer/formatter/attribute_test.exs:28
     Assertion with == failed
     code:  assert {"example", "foobar"} == results
     left:  {"example", "foobar"}
     right: {"example", %{"bar" => "bar", "foo" => "foo"}}
     stacktrace:
       test/ja_serializer/formatter/attribute_test.exs:35: (test)



  2) test allows overriding for struct formatting (JaSerializer.Formatter.AttributeTest)
     test/ja_serializer/formatter/attribute_test.exs:20
     ** (Protocol.UndefinedError) protocol Enumerable not implemented for %JaSerializer.Formatter.AttributeTest.Example{bar: "bar", foo: "foo"}
     code: JaSerializer.Formatter.format(%@attr{
     stacktrace:
       (elixir) lib/enum.ex:1: Enumerable.impl_for!/1
       (elixir) lib/enum.ex:141: Enumerable.reduce/3
       (elixir) lib/enum.ex:2979: Enum.reduce/3
       (ja_serializer) lib/ja_serializer/formatter/attribute.ex:5: JaSerializer.Formatter.JaSerializer.Builder.Attribute.format/1
       test/ja_serializer/formatter/attribute_test.exs:22: (test)
```

I get the same failures when running off your first commit (b67be5d334ca537697ef834f6f6978d37611da7a).

**Comment by broodfusion** [comment]
Thanks for the comments.
I've forked a new master branch and now tests were passing. I suppose I'll close this PR and open another one to start fresh.

**Comment by beerlington** [comment]
@broodfusion thanks for putting this together! I know it's been a long time, but I'm shooting to get a release out before the end of the year. One of the things that has slowed down review has been the formatting change in the diff making it harder to see what has actually changed. I don't want you to have to revert that, but would like to keep PRs scoped to just the changes needed. 

I think what makes sense is for me to just run the formatter on master, and then have you rebase once that's in. I'll ping you when it's ready!

**Comment by beerlington** [comment]
@broodfusion master is good to go now

**Comment by broodfusion** [comment]
Closing PR
An updated one has been submitted here: https://github.com/vt-elixir/ja_serializer/pull/302

**Pull request closed without merging**